### PR TITLE
Quad 79

### PR DESCRIPTION
**Pull request by nischalsamji**
Complete Code Changes for Quadriga Text Management

### CODE REVIEW

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TextHandlerDTO.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.Table;
+
+public class TextHandlerDTO {
+
+    @Entity
+    @Table(name = "tbl_textfiles")
+    public class TextFileEntity implements Serializable {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TextHandlerDTO.java]: why do you have a class in a class here?

File: Quadriga/src/main/webapp/WEB-INF/views/requestaccount.jsp [hunk 2]
```diff
@@ -30,4 +31,38 @@
...
+			<td>Password: </td><td><form:password  path="password" /></td>
+		</tr>
+		<tr>
+			<td>Repeat password: </td><td><form:password path="repeatedPassword" /></td>
+		</tr>
+		
+		</table>
+		<input type="submit" value="Create Account">
+	</form:form>
+>>>>>>> branch 'QUAD-79' of https://github.com/diging/quadriga.git
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/requestaccount.jsp]: make sure to resolve all conflicts

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TextHandlerDTO.java (hunk 1 above)
Comment (nischalsamji) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TextHandlerDTO.java]: guess this DTO is not required, deleted it.

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
The pop-up for the text needs a scrollbar:
image

image

**Comment by jdamerow** [comment]
Make sure to use utf-8 for the encoding. There are some question marks which probably means there are some encoding issues.

**Comment by jdamerow** [comment]
Also, make sure to keep new lines and show them correctly when displaying the text (there were two newlines after the first "Beatrice Mintz".

**Comment by jdamerow** [comment]
otherwise looks great!

**Comment by nischalsamji** [comment]
Updated the Modal View

**Comment by nischalsamji** [comment]
Using UTF-8, changed it to encoding used in jsp pages

**Comment by jdamerow** [comment]
I'm getting the following exception:

```
edu.asu.spring.quadriga.exceptions.QuadrigaException: could not execute statement; SQL [n/a]; constraint [null]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
    org.springframework.orm.hibernate4.SessionFactoryUtils.convertHibernateAccessException(SessionFactoryUtils.java:163)
    org.springframework.orm.hibernate4.HibernateTransactionManager.convertHibernateAccessException(HibernateTransactionManager.java:730)
    org.springframework.orm.hibernate4.HibernateTransactionManager.doCommit(HibernateTransactionManager.java:592)
    org.springframework.transaction.support.AbstractPlatformTransactionManager.processCommit(AbstractPlatformTransactionManager.java:761)
    org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:730)
    org.springframework.transaction.interceptor.TransactionAspectSupport.commitTransactionAfterReturning(TransactionAspectSupport.java:485)
    org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:291)
    org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96)
    org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
    org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:208)
    com.sun.proxy.$Proxy6757.saveTextFile(Unknown Source)
    edu.asu.spring.quadriga.web.workspace.AddTextController.saveTextFileForm(AddTextController.java:121)
    edu.asu.spring.quadriga.web.workspace.AddTextController$$FastClassBySpringCGLIB$$f566bb1f.invoke(<generated>)
    org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
    org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:720)
    org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
    org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:85)
    edu.asu.spring.quadriga.aspects.AccessAspect.chkAuthorization(AccessAspect.java:83)
    sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    java.lang.reflect.Method.invoke(Method.java:497)
```

try copying/pasting this text: https://embryo.asu.edu/pages/anne-laura-dorinthea-mclaren-1927-2007

**Comment by nischalsamji** [comment]
I am able to do it, no exceptions
screen shot 2016-05-02 at 12 25 53 pm

**Comment by jdamerow** [comment]
Hm, weird, the project id was set as primary key. Fixed it, but Is till get several question marks:
image

**Comment by nischalsamji** [comment]
Is the browser Safari? I am testing on chrome, unable to see the question marks.
screen shot 2016-05-02 at 12 43 16 pm

Edit: Works fine on Safari too.

**Comment by jdamerow** [comment]
no, i'm on chrome too.

**Comment by jdamerow** [comment]
check here: http://diging.asu.edu:8081/quadriga-review/auth/workbench/workspace/workspacedetails/62dd6b36-93d6-46b3-a3f0-2c21bc20d910

**Comment by nischalsamji** [comment]
Yup... On server am seeing question marks, unable to see them on my local machine.

**Comment by jdamerow** [comment]
retest this please

**Comment by nischalsamji** [comment]
Fixed, Can you check this once ?

**Comment by nischalsamji** [comment]
Fixed the "No networks found message"

**Comment by jdamerow** [comment]
filename in modal is still missing

**Comment by nischalsamji** [comment]
Completed.

**Pull request merged**